buildkite/bazel-central-registry/bcr_presubmit.py

```diff
@@ -140,13 +170,85 @@ def create_test_repo(module_name, module_version, task):
     return root
 
 
-def run_test(repo_location, module_name, module_version, task):
+def download(url, file):
+    with urllib.request.urlopen(url) as response:
+        with open(file, "wb") as f:
+            f.write(response.read())
+
+
+def read(path):
+    with open(path, "r") as file:
+        return file.read()
+
+
+def load_source_json(module_name, module_version):
+    source_json = get_source_json(module_name, module_version)
+    with open(source_json, "r") as json_file:
+        return json.load(json_file)
+
+
+def apply_patch(work_dir, patch_strip, patch_file):
+    # Requries patch to be installed, this is true for all Bazel CI VMs, including Windows VMs.
+    subprocess.run(
+        ["patch", "-p%d" % patch_strip, "-i", patch_file], shell=False, check=True, env=os.environ, cwd=work_dir
+    )
+
+
+def prepare_test_module_repo(module_name, module_version, task):
+    """Prepare the test module repo and the presubmit yml file it should use"""
+    bazelci.print_collapsed_group(":information_source: Prepare test module repo")
+    root = get_root_dir(module_name, module_version, task, is_test_module = True)
+    source = load_source_json(module_name, module_version)
+
+    # Download and unpack the source archive to ./output
+    archive_url = source["url"]
+    archive_file = root.joinpath(archive_url.split("/")[-1])
+    output_dir = root.joinpath("output")
+    bazelci.eprint("* Download and unpack %s\n" % archive_url)
+    download(archive_url, archive_file)
+    shutil.unpack_archive(str(archive_file), output_dir)
```

Comment:
At present, we lose the permissions on the extracted files, per python/cpython#59999.